scarlet2/frame.py

```diff
@@ -38,75 +40,92 @@ def pixel_size(self):
             return 1
 
     def get_pixel(self, sky_coord):
-        """Get the pixel coordinate from a world coordinate
+        """Get the sky coordinates from a world coordinate
 
         Parameters
         ----------
-        sky_coord: tuple, array
+        sky_coord: SkyCoord, or list of Skycoords
             Coordinates on the sky
+
+        Returns
+        ---------
+        pixel coordinates in the model frame
         """
-        sky = jnp.asarray(sky_coord, dtype=jnp.float32).reshape(-1, 2)
+        assert self.wcs is not None
+        wcs_ = self.wcs.celestial # only use celestial portion
 
-        if self.wcs is not None:
-            wcs_ = self.wcs.celestial  # only use celestial portion
-            pixel = jnp.array(wcs_.world_to_pixel_values(sky)).reshape(-1, 2)
-            # y/x instead of x/y:
-            pixel = jnp.flip(pixel, axis=-1)
+        if isinstance(sky_coord, list):
+            assert all(isinstance(coord, SkyCoord) for coord in sky_coord)
+            pixel = jnp.asarray([coord.to_pixel(wcs_) for coord in sky_coord],
+                                 dtype="float32")     
         else:
-            pixel = sky
-
-        if pixel.size == 2:  # only one coordinate pair
-            return pixel[0]
+            assert isinstance(sky_coord, SkyCoord)
+            pixel = jnp.asarray(sky_coord.to_pixel(wcs_), dtype="float32")
+        
         return pixel
-
-    def get_sky_coord(self, pixel):
+    
+    def get_sky_coord(self, pixels):
         """Get the sky coordinate from a pixel coordinate
 
         Parameters
         ----------
-        pixel: tuple, array
+        pixel: array of pixel coordinates 
             Coordinates in the pixel space
-        """
-        pix = jnp.asarray(pixel, dtype=np.float64).reshape(-1, 2)
 
-        if self.wcs is not None:
-            wcs = self.wcs.celestial  # only use celestial portion
-            # x/y instead of y/x:
-            pix = jnp.flip(pix, axis=-1)
-            sky = jnp.array(wcs.pixel_to_world_values(pix))
-        else:
-            sky = pix
+        Returns
+        ----------
+        astropy.coordinates.SkyCoord
+        """
+        pixels = pixels.reshape(-1, 2)
 
-        if sky.size == 2:
-            return sky[0]
-        return sky
+        assert self.wcs is not None
+        wcs = self.wcs.celestial # only use celestial portion
+        sky_coord = [SkyCoord.from_pixel(pixel[0], pixel[1], wcs) for pixel in pixels]
```

Comment:
Same here, `SkyCoord` should transform ndarrays in one go: https://docs.astropy.org/en/stable/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord.from_pixel